### PR TITLE
Feat/z-carousel

### PR DESCRIPTION
**Pull request by StefanoMarcolini97**
# Feature - Z-Carousel - add new component

----

### Priority

- [ ] 1 - Highest
- [ ] 2 - High
- [x] 3 - Medium
- [ ] 4 - Low
- [ ] 5 - Lowest
- [ ] 6 - Not a Priority

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [x] Component (add a Component as approved by Design System)
- [ ] Docs (add documentation)
- [ ] Chore (changes that adds small enhancement)
- [ ] Breaking (fix or feature that would cause existing functionality to not work as expected)

### Abstract
[link](https://app.abstract.com/projects/fd370780-e659-11e8-99dc-0d08537c5fde/branches/86a3c8c1-879e-45db-89c8-6b988e340714/commits/latest/files/9f586aa0-bd5b-4c12-9541-aed2e98a7a9c/layers/73601D1F-4173-4B38-8AE3-4E0C60EFE70E?collectionId=0c1fd332-ecb9-4396-b47f-d839bea18fd6&collectionLayerId=3587b245-8d7d-42d9-ac8f-d39a67a2edd2)

### Note

----

## Component and Fix approval flow to Master branch

A Pull Request to be merged must have two approvals, one from a member of the product team who developed it and another from a member of the dst dev team other than the team representative.

### CODE REVIEW

**Comment by DarioSacco** [comment]
nota per il futuro: non usare i codici delle issue di prodotto (tipo `CLAV-275`) nei nomi dei branch e commit di questo repo 😬

**Review by DarioSacco — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@zanichelli/albe-web-components",
-  "version": "2.37.0",
+  "version": "2.37.1-CLAV-275",
```
[the remark this answers is not in the document]
Comment (StefanoMarcolini97) [reply package.json:3]: Refuso dopo una prova di integrazione su registro, sistemato

File: src/components/z-carousel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  componentDidRender() {
+    const children = this.host.children[0].children;
+    for (let i = 0; i < children.length - 1; i++) {
+      children[i].setAttribute("gutter", `${this.gutter}`);
+    }
+  }
+
+  render() {
+    return (
+      <Host>
```
[the remark this answers is not in the document]
Comment (StefanoMarcolini97) [reply src/components/z-carousel/index.tsx:23]: Ok chiaro, lo rimuovo

File: src/components/z-carousel/z-carousel-item/index.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  shadow: false,
+  scoped: true,
+})
+export class ZCarouselItem {
+  /** [optional] gutter between the slide items */
+  @Prop() gutter?: number = 0;
+
+  render() {
+    return (
+      <li class="carousel-item" style={{ paddingRight: `${this.gutter}px` }}>
```
[the remark this answers is not in the document]
Comment (StefanoMarcolini97) [reply src/components/z-carousel/z-carousel-item/index.tsx:15]: In effetti le due cose sono equivalenti, avevamo creato lo z-carousel-item prendendo ad esempio lo z-list

File: src/components/z-carousel/index.stories.mdx [hunk 4]
```diff
@@ -244,6 +244,8 @@ It was conceived as a list of elements to respect the rules of accessibility. (h
...
 ## Usage
 
-One `z-carousel`component wraps `N` `z-carousel-item`.
-Any type of element can be passed to the `z-carousel-item`.
-The size of the `z-carousel-item` fits to the size of the children
+One `z-carousel`component wraps `N` `li`.
+
+Any type of element must be passed into a `li` element.
+
+The size of the carousel children fits to the size of the children
```
Comment (DarioSacco) [inline src/components/z-carousel/index.stories.mdx:251]: non ho capito che intendi con questa frase 🤔

File: src/components/z-carousel/index.tsx [hunk 5]
```diff
@@ -1,30 +1,24 @@
...
-    for (let i = 0; i < children.length - 1; i++) {
-      children[i].setAttribute("gutter", `${this.gutter}`);
-    }
-  }
-
   render() {
     return (
-      <Host>
-        <ul class="carousel-item-container">
+      <div>
```
Comment (DarioSacco) [inline src/components/z-carousel/index.tsx:17]: come mai questo `div` e non direttamente `ul`?

**Review by DarioSacco — APPROVED** [review]

**Pull request merged**